caboose-util/src/main.rs

```diff
@@ -0,0 +1,38 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2023 Oxide Computer Company
+
+use anyhow::{anyhow, bail, Context, Result};
+use hubtools::RawHubrisArchive;
+use tlvc_text::{Piece, Tag};
+
+fn main() -> Result<()> {
+    let mut args = std::env::args().skip(1);
+    match args.next().context("subcommand required")?.as_str() {
+        "read-version" => {
+            let archive = RawHubrisArchive::load(
+                &args.next().context("path to hubris archive required")?,
+            )?;
+            let caboose = archive.read_caboose()?;
```

Comment:
I think you could use [`caboose.version()`](https://github.com/oxidecomputer/hubtools/blob/50de6c304df47dfaa19d15c170f4d7980341d2af/hubtools/src/caboose.rs#L43-L45) at this point instead of manually decoding the TLVC content.

Reply:
how did i miss this!